rhel/vex/parser.go

```diff
@@ -501,11 +546,32 @@ func cvssVectorFromScore(sc *csaf.Score) (vec string, err error) {
 
 // CreatePackageKey creates a unique key to describe an arch agnostic
 // package for deduplication purposes.
-// i.e. AppStream-8.2.0.Z.TUS:python3-idle-0:3.6.8-24.el8_2.2
-func createPackageKey(repo, name, fixedIn string) string {
+// i.e. AppStream-8.2.0.Z.TUS:a_module:python3-idle-0:3.6.8-24.el8_2.2
+func createPackageKey(repo, mod, name, fixedIn string) string {
 	// The other option here is just to use repo + PURL string
 	// w/o the qualifiers I suppose instead of repo + NEVR.
-	return repo + ":" + name + "-" + fixedIn
+	return repo + ":" + mod + ":" + name + "-" + fixedIn
+}
+
+func createPackageModule(p *csaf.Product) (string, error) {
+	var modName string
+	modPURLHelper, ok := p.IdentificationHelper["purl"]
+	if !ok {
+		return modName, nil
+	}
+	purl, err := packageurl.FromString(modPURLHelper)
+	if err != nil {
+		return "", err
+	}
+	if purl.Type != "rpmmod" {
+		return "", fmt.Errorf("invalid RPM module PURL: %q", purl.String())
+	}
+	if version, _, found := strings.Cut(purl.Version, ":"); found {
+		modName = purl.Name + ":" + version
+	} else {
```

Comment:
@RTann adding this clause as some of the PURLs don't conform to a standard format and will be potentially updated:
Conventional: `pkg:rpmmod/redhat/postgresql@13:8080020231114105206:63b34585`
Unconventional: `pkg:rpmmod/redhat/postgresql:15/postgresql`
Possible update for unconventional format: `pkg:rpmmod/redhat/postgresql@15`

This is being looked at by RH prodsec.

Reply:
Based on https://go.dev/play/p/PamAwGzH8r0 I think you can just do

```
version, _, _ := strings.Cut(purl.Version, ":")
return purl.Name + ":" + version, nil
```

Reply:
> This is being looked at by RH prodsec.

Is this blocked, then?

Reply:
I'm not sure there is a point in editing it until we get a reply about whether this is:

1. A bug and will be fixed
2. A bug and won't be immediately fixed
3. Not a but and won't be changed

If we don't get that answer soon I think we just account for the conventional way and possibly modify in the future.

Reply:
It's been a couple of days with no reply, going to change to just account for conventional `rpmmod` PURLs